### DISCUSSION TITLE
Smart Energy Hot Water Integration

### DISCUSSION THREAD

**Discussion by dominikamann** [WillCodeForCats/solaredge-modbus-multi#222] [Ideas]
Hi,

i use the https://www.solaredge.com/de/products/smart-energy/smart-energy-hot-water#/ component to use the surplus energy by my own. The energy is used for heating the water inside of my hygienic storage. This works quite well I don’t need an other heater during the summer time. But in winter/spring on sunny days often the battery will charged first and after this the component gets the energy. This is in many cases very inefficient. My plan is to steer the heating rod by some intelligent ha automations. E.g is the battery > 60% and still more than 1kw solar power then turn the heater on. This leads to a slower battery charge but better usage of the energy over the day. Because the battery always use the full available power (up to 5kwh) to charge. In some cases the battery is full at 2am and then the heating rod will turn on. More efficient would be battery full at 5pm and meanwhile using the power to heat. 

Do you have any plans to integrate the component or an idea how to achieve this?

Thank you very much!

**Comment by WillCodeForCats** [comment]
No, other devices can't be added: not unless SolarEdge expands their modbus support to include them. If they do I'm happy to integrate them in the future.

Otherwise you could probably do some automatons with external metering on the water heater. Shelly has energy metering models: https://www.shelly.cloud

Also see: https://github.com/WillCodeForCats/solaredge-modbus-multi/discussions/19

**Reply by dominikamann** [reply]
Thank you for the quick response. Ok so it is currently not part of the Solar Edge modbus functionality. That's sad. I will write to Solar Edge may they will do something. Yes, I could use a Shelly between the heating rod and the smart-heating-component. Then switching the energy priorisation to "heating first" (instead of battery). This is a way to control it by my own. I hoped to find a more elegant way. 

![image](https://user-images.githubusercontent.com/29973737/210180310-c7710161-d1e4-44e4-88ea-661bd8fdc20f.jpeg)

**Comment by Homebrew09** [comment]
I am interested in this as well. I went in contact to solaredge support as well and they send me infos for TCP, Modbus and WEB API:
https://knowledge-center.solaredge.com/sites/kc/files/se_monitoring_api.pdf
https://knowledge-center.solaredge.com/sites/kc/files/sunspec-implementation-technical-note.pdf
I could not find any infos for the water controller in both of them.  So I went back to them and ask the how to access, but right now, I did not get an answer.
My usecase is, that I would like to enable the water controller in the morning if battery is charged enough...
